pkg/util/flags.go

```diff
@@ -49,3 +50,23 @@ func WarnDeprecatedConfig(flagName string, logger log.Logger) {
 	flagext.DeprecatedFlagsUsed.Inc()
 	level.Warn(logger).Log("msg", fmt.Sprintf("the configuration parameter -%s is deprecated and will be soon removed from Mimir", flagName))
 }
+
+// ParseFlagsAndArguments calls Parse() on the input flag.FlagSet and returns the parsed arguments.
+func ParseFlagsAndArguments(f *flag.FlagSet) ([]string, error) {
+	err := f.Parse(os.Args[1:])
+	return f.Args(), err
+}
+
+// ParseFlags calls Parse() on the input flag.FlagSet and enforces no arguments have been parsed.
+// This utility should be called whenever we only expect CLI flags but no arguments.
+func ParseFlags(f *flag.FlagSet) error {
```

Comment:
[nit] Shall we better signal the purpose in the name of the function? `ParseFlagsWithoutArguments` or `ParseFlagsOnly`.

Reply:
Renamed to `ParseFlagsWithoutArguments`.